### ISSUE TITLE
Container VM process exits non-zero but is not caught by the portlayer

### ISSUE THREAD

**Issue by jzt** [vmware/vic#3728]
Seen in a CI test:

A `/bin/top` container is created without `-t`, causing `top` to exit with status `1` because no `TERM` is set. This causes the container to shutdown, but the non-zero exit status is not bubbled up, and a CI test passes when it should have failed.

Excerpt from tether.debug:
```
time="2017-01-18T23:58:18Z" level=info msg="Launching process for session 933e252ce6fd063388d571849dc52f044a892c67becf175dfbcb63b6c1a5da52" 
time="2017-01-18T23:58:18Z" level=debug msg="[BEGIN] [github.com/vmware/vic/lib/tether.(*tether).launch:502] launching session 933e252ce6fd063388d571849dc52f044a892c67becf175dfbcb63b6c1a5da52" 
time="2017-01-18T23:58:18Z" level=debug msg="[BEGIN] [main.(*attachServerSSH).run:285] main attach server loop" 
time="2017-01-18T23:58:18Z" level=info msg="Trying to establish a connection" 
time="2017-01-18T23:58:18Z" level=info msg="opening ttyS0 for backchannel" 
time="2017-01-18T23:58:18Z" level=info msg="setting terminal to raw mode" 
time="2017-01-18T23:58:18Z" level=debug msg="s = &terminal.State{termios:syscall.Termios{Iflag:0x500, Oflag:0x5, Cflag:0xcbd, Lflag:0x8a3b, Line:0x0, Cc:[32]uint8{0x3, 0x1c, 0x7f, 0x15, 0x4, 0x0, 0x1, 0x0, 0x11, 0x13, 0x1a, 0x0, 0x12, 0xf, 0x17, 0x16, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, Pad_cgo_0:[3]uint8{0x0, 0x0, 0x0}, Ispeed:0x0, Ospeed:0x0}}" 
time="2017-01-18T23:58:18Z" level=info msg="creating raw connection from ttyS0 (fd=7)" 
time="2017-01-18T23:58:18Z" level=debug msg="[BEGIN] [main.backchannel:231] establish tether backchannel" 
time="2017-01-18T23:58:18Z" level=debug msg="HandshakeServer: Waiting for incoming syn request..." 
time="2017-01-18T23:58:18Z" level=debug msg="Resolved top to /usr/bin/top" 
time="2017-01-18T23:58:18Z" level=debug msg="[BEGIN] [github.com/vmware/vic/lib/tether.establishNonPty:248] initializing nonpty handling for session 933e252ce6fd063388d571849dc52f044a892c67becf175dfbcb63b6c1a5da52" 
time="2017-01-18T23:58:18Z" level=debug msg="Setting StdoutPipe" 
time="2017-01-18T23:58:18Z" level=debug msg="Setting StderrPipe" 
time="2017-01-18T23:58:18Z" level=debug msg="[ END ] [github.com/vmware/vic/lib/tether.establishNonPty:248] [24.098812ms] initializing nonpty handling for session 933e252ce6fd063388d571849dc52f044a892c67becf175dfbcb63b6c1a5da52" 
time="2017-01-18T23:58:18Z" level=debug msg="Launched command with pid 222" 
time="2017-01-18T23:58:18Z" level=debug msg="[ END ] [github.com/vmware/vic/lib/tether.(*tether).launch:502] [138.53831ms] launching session 933e252ce6fd063388d571849dc52f044a892c67becf175dfbcb63b6c1a5da52" 
TERM environment variable not set.
time="2017-01-18T23:58:18Z" level=debug msg="[0xc4201bd920] write \"TERM environment variable not set.\\n\" to 2 writers (err: <nil>)" 
time="2017-01-18T23:58:18Z" level=debug msg="Writer goroutine for stdout returned: %!s(<nil>)" 
time="2017-01-18T23:58:18Z" level=debug msg="Writer goroutine for stdout exiting" 
time="2017-01-18T23:58:18Z" level=debug msg="Writer goroutine for stderr returned: %!s(<nil>)" 
time="2017-01-18T23:58:18Z" level=debug msg="Writer goroutine for stderr exiting" 
time="2017-01-18T23:58:18Z" level=debug msg="Inspecting children with status change" 
time="2017-01-18T23:58:18Z" level=debug msg="Reaped process 222, return code: 1" 
```

output.log:
```
TERM environment variable not set.
```
The test passes because the docker client exits with status 0, meaning the portlayer back through to the persona never saw the error code from the container VM process.

A subsequent test involving removal of the container failed, as `stop` was not able to stop an already powered off VM (see #3718).

Log bundle:  https://github.com/vmware/vic/files/717182/integration_logs_7580_d4eb6551f8cdc5b8d90e0e2f8015125e7f705a37.zip

**Comment by hickeng** [comment]
If I'm correct, this is an artifact of us failing to have clear status reporting for process launch status. This status reporting issue is to be addressed as part of the #3618 work, but may need to be partially addressed for this issue and #5532

**Comment by hickeng** [comment]
It's quite possible that this was fixed recently - in the #5532 merge dealing with exit code handling.

**Comment by mdubya66** [comment]
Closing as fixed as part of #5532. We can re-open if seen again.
